src/responder.ts

```diff
@@ -417,57 +382,92 @@ export class EthereumDedicatedResponder extends EthereumResponder {
     }
 }
 
-
 /**
  * Responsible for handling the business logic of the Responders.
  */
 // TODO: This is a mock class and only correctly handles one active response.
 //       Should add a pool of wallets to allow concurrent responses.
 
-export class EthereumResponderManager {
+export class EthereumResponderManager extends StartStopService {
+    // Waiting time before throwing an error if no new blocks are received, in milliseconds
+    public static readonly WAIT_TIME_FOR_NEW_BLOCK = 120 * 1000;
+
     private provider: ethers.providers.Provider;
     private responders: Set<EthereumResponder> = new Set();
     private gasPolicy: IGasPolicy;
 
-    constructor(private readonly signer: ethers.Signer) {
+    private blockTimeoutDetector: BlockTimeoutDetector;
+    private confirmationObserver: ConfirmationObserver;
+
+    constructor(private readonly signer: ethers.Signer, blockCache: BlockCache, blockProcessor: BlockProcessor) {
+        super("EthereumResponderManager");
+
         if (!signer.provider) throw new ArgumentError("The given signer is not connected to a provider");
 
         this.provider = signer.provider;
 
         this.gasPolicy = new DoublingGasPolicy(this.provider);
+
+        this.blockTimeoutDetector = new BlockTimeoutDetector(blockProcessor, 120 * 1000);
```

Comment:
These should be detected, and EthereumResponderManager shouldnt be a startstopservice

Reply:
Done in fcfe337.